mono/mini/method-to-ir.c

```diff
@@ -8592,72 +8609,26 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				}
 			}
 			ins = (MonoInst*)mini_emit_calli_full (cfg, fsig, sp, addr, NULL, NULL, tailcall);
-
```

Comment:
I don't think we should goto between switch cases, its really confusing. Just keep the duplicate code.

Reply:
I think it is too much duplicated code. It was less before, and it has incorrectly diverged even then. Now it is larger, though maybe could be dialed back.

Ultimately I think the common tail becomes a function and then can call it from two places.